src/main/java/appeng/util/helpers/ItemComparisonHelper.java

```diff
@@ -0,0 +1,443 @@
+		{
+			switch( id )
+			{
+				case 10:
+				{
+					final NBTTagCompound ctA = (NBTTagCompound) left;
+					final NBTTagCompound ctB = (NBTTagCompound) right;
+
+					final Set<String> cA = ctA.getKeySet();
+					final Set<String> cB = ctB.getKeySet();
+
+					if( cA.size() != cB.size() )
+					{
+						return false;
+					}
+
+					for( final String name : cA )
+					{
+						final NBTBase tag = ctA.getTag( name );
+						final NBTBase aTag = ctB.getTag( name );
+						if( aTag == null )
+						{
+							return false;
+						}
+
+						if( !isNbtTagEqual( tag, aTag ) )
+						{
+							return false;
+						}
+					}
+
+					return true;
+				}
+
+				case 9: // ) // A instanceof NBTTagList )
+				{
+					final NBTTagList lA = (NBTTagList) left;
+					final NBTTagList lB = (NBTTagList) right;
+					if( lA.tagCount() != lB.tagCount() )
+					{
+						return false;
+					}
+
+					final List<NBTBase> tag = tagList( lA );
+					final List<NBTBase> aTag = tagList( lB );
+					if( tag.size() != aTag.size() )
+					{
+						return false;
+					}
+
+					for( int x = 0; x < tag.size(); x++ )
+					{
+						if( aTag.get( x ) == null )
+						{
+							return false;
+						}
+
+						if( !isNbtTagEqual( tag.get( x ), aTag.get( x ) ) )
+						{
+							return false;
+						}
+					}
+
+					return true;
+				}
+
+				case 1: // ( A instanceof NBTTagByte )
+					return ( (NBTPrimitive) left ).getByte() == ( (NBTPrimitive) right ).getByte();
+
+				case 4: // else if ( A instanceof NBTTagLong )
+					return ( (NBTPrimitive) left ).getLong() == ( (NBTPrimitive) right ).getLong();
+
+				case 8: // else if ( A instanceof NBTTagString )
+					return ( (NBTTagString) left ).getString().equals( ( (NBTTagString) right ).getString() ) || ( (NBTTagString) left ).getString().equals( ( (NBTTagString) right ).getString() );
+
+				case 6: // else if ( A instanceof NBTTagDouble )
+					return ( (NBTPrimitive) left ).getDouble() == ( (NBTPrimitive) right ).getDouble();
+
+				case 5: // else if ( A instanceof NBTTagFloat )
+					return ( (NBTPrimitive) left ).getFloat() == ( (NBTPrimitive) right ).getFloat();
+
+				case 3: // else if ( A instanceof NBTTagInt )
+					return ( (NBTPrimitive) left ).getInt() == ( (NBTPrimitive) right ).getInt();
+
+				default:
+					return left.equals( right );
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Unordered hash of NBT Data, used to work thought huge piles fast, but ignores the order just in case MC
+	 * decided to change it... WHICH IS BAD...
+	 */
+	public int createUnorderedNbtHash( final NBTBase nbt )
+	{
+		// same type?
+		int hash = 0;
+		final byte id = nbt.getId();
+		hash += id;
+		switch( id )
+		{
+			case 10:
+			{
+				final NBTTagCompound ctA = (NBTTagCompound) nbt;
+
+				final Set<String> cA = ctA.getKeySet();
+
+				for( final String name : cA )
+				{
+					hash += name.hashCode() ^ createUnorderedNbtHash( ctA.getTag( name ) );
+				}
+
+				return hash;
+			}
+
+			case 9: // ) // A instanceof NBTTagList )
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 13 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1151)